src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -403,27 +419,39 @@ class DemuxFastqs
     val isUnmatched = sample.sampleName == UnmatchedSampleId
     val prefix = toSampleOutputPrefix(sample, isUnmatched, illuminaStandards, output, this.unmatched)
 
-    if (this.outputFastqs) {
-      new FastqRecordWriter(prefix, pairedEnd, illuminaStandards)
+    val writers = new ListBuffer[DemuxWriter]()
+
+    this.outputType match {
```

Comment:
These matches seem to me like they're be more at home as:
`if (this.outputType == FastqOnly || this.outputType == Both)` since you don't use the matched value _and_ you then have add a second default case.

You could simplify a little further by having the `OutputType` enum have a `.producesFastq` and `.producesBam` pair of values ;)